exp/api/v1alpha3/azuremachinepool_types.go

```diff
@@ -0,0 +1,152 @@
+		// for machine interpretation.
+		//
+		// This field should not be set for transitive errors that a controller
+		// faces that are expected to be fixed automatically over
+		// time (like service outages), but instead indicate that something is
+		// fundamentally wrong with the MachinePool's spec or the configuration of
+		// the controller, and that manual intervention is required. Examples
+		// of terminal errors would be invalid combinations of settings in the
+		// spec, values that are unsupported by the controller, or the
+		// responsible controller itself being critically misconfigured.
+		//
+		// Any transient errors that occur during the reconciliation of MachinePools
+		// can be added as events to the MachinePool object and/or logged in the
+		// controller's output.
+		// +optional
+		FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
+
+		// ErrorMessage will be set in the event that there is a terminal problem
+		// reconciling the MachinePool and will contain a more verbose string suitable
+		// for logging and human consumption.
+		//
+		// This field should not be set for transitive errors that a controller
+		// faces that are expected to be fixed automatically over
+		// time (like service outages), but instead indicate that something is
+		// fundamentally wrong with the MachinePool's spec or the configuration of
+		// the controller, and that manual intervention is required. Examples
+		// of terminal errors would be invalid combinations of settings in the
+		// spec, values that are unsupported by the controller, or the
+		// responsible controller itself being critically misconfigured.
+		//
+		// Any transient errors that occur during the reconciliation of MachinePools
+		// can be added as events to the MachinePool object and/or logged in the
+		// controller's output.
+		// +optional
+		FailureMessage *string `json:"failureMessage,omitempty"`
+	}
+
+	// +kubebuilder:object:root=true
+	// +kubebuilder:subresource:status
+	// +kubebuilder:printcolumn:name="Replicas",type="string",JSONPath=".status.replicas",description="AzureMachinePool replicas count"
+	// +kubebuilder:printcolumn:name="Ready",type="string",JSONPath=".status.ready",description="AzureMachinePool replicas count"
+	// +kubebuilder:printcolumn:name="Cluster",type="string",priority=1,JSONPath=".metadata.labels.cluster\\.x-k8s\\.io/cluster-name",description="Cluster to which this AzureMachinePool belongs"
+	// +kubebuilder:printcolumn:name="MachinePool",type="string",priority=1,JSONPath=".metadata.ownerReferences[?(@.kind==\"MachinePool\")].name",description="MachinePool object to which this AzureMachinePool belongs"
+	// +kubebuilder:printcolumn:name="VMSS ID",type="string",priority=1,JSONPath=".spec.providerID",description="Azure VMSS ID"
+	// +kubebuilder:printcolumn:name="VM Size",type="string",priority=1,JSONPath=".spec.template.vmSize",description="Azure VM Size"
```

Comment:
does it make sense to have ProvisioningState here?

Reply:
Happy to add that. Again, I had just followed what we were displaying for VM.

Reply:
pretty sure I added provisioning state for VMs, it's called "vmState" though

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/api/v1alpha3/azuremachine_types.go#L118

Reply:
Indeed you had. Oops, I missed that one. Adding it...